sphinx_gallery/tests/test_full.py

```diff
@@ -169,8 +203,8 @@ def test_embed_links_and_styles(sphinx_app):
     # highlight language
     fname = op.join(src_dir, 'auto_examples', 'plot_numpy_matplotlib.rst')
     assert op.isfile(fname)
-    with open(fname, 'rb') as fid:
-        rst = fid.read().decode('utf-8')
+    with codecs.open(fname, 'r', 'utf-8') as fid:
```

Comment:
is `codecs` better than just `open`? I've never seen that used before

Reply:
It's the preferred method in SG (look around the codebase and you'll see it a lot). Everywhere else in the Python ecosystem I tend to see `with open(x, 'rb') as fid: fid.read().decode('utf-8')` but this was frowned upon when I did it this way a while ago.